### PR TITLE
Fix #3498: Add CI check to ensure license text is never checked into Git

### LINKED ISSUE #3498: Add CI check to ensure that the actual license texts can’t be accidentally checked into Git
We never want to contain actual license texts in our production code and thus a CI check needs to be added that verifies this.

### PR DESCRIPTION
**Pull request by prayutsu**
## Explanation

Fixes #3498  
Added a new script that will run on CI and verify that the version of third_party_dependencies.xml generated by GenerateLicenseTexts.kt is never checked into Git.

Screenshot -
Screenshot from 2021-08-01 20-57-14

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Comment by prayutsu** [comment]
@BenHenning @rt4914 PTAL.

**Review by rt4914 — APPROVED** [review]
LGTM. Thanks.

File: scripts/src/java/org/oppia/android/scripts/ci/LicenseTextsCheck.kt
Comment (BenHenning) [inline scripts/src/java/org/oppia/android/scripts/ci/LicenseTextsCheck.kt]: No need to have ``@param`` in this way. I suggest defining the parameters explicitly like we do in the other scripts since you're describing the CLI for a command rather than the arguments for a function being called.

File: scripts/src/java/org/oppia/android/scripts/ci/LicenseTextsCheck.kt
Comment (BenHenning) [inline scripts/src/java/org/oppia/android/scripts/ci/LicenseTextsCheck.kt]:
```suggestion
    throw Exception("Too few arguments passed")
```

File: scripts/src/java/org/oppia/android/scripts/ci/LicenseTextsCheck.kt
Comment (BenHenning) [inline scripts/src/java/org/oppia/android/scripts/ci/LicenseTextsCheck.kt]:
```suggestion
  if (comment !in xmlContent.contains) {
```

More idiomatic version.

File: scripts/src/java/org/oppia/android/scripts/ci/LicenseTextsCheck.kt
Comment (BenHenning) [inline scripts/src/java/org/oppia/android/scripts/ci/LicenseTextsCheck.kt]:
```suggestion
    throw Exception("License texts potentially checked into VCS")
```

Technically it's not guaranteed since you aren't verifying it (someone might have just removed or changed the warning, for example).

File: scripts/src/java/org/oppia/android/scripts/ci/LicenseTextsCheck.kt
Comment (BenHenning) [inline scripts/src/java/org/oppia/android/scripts/ci/LicenseTextsCheck.kt]:
```suggestion
  val xmlContent = thirdPartyDepsXml.readText()
```

Might this work, instead?

File: scripts/src/java/org/oppia/android/scripts/ci/LicenseTextsCheck.kt
Comment (BenHenning) [inline scripts/src/java/org/oppia/android/scripts/ci/LicenseTextsCheck.kt]: I suggest introducing a scripts.license subpackage for this & your earlier PR, instead. The CI subpackage is for scripts needed to run the CI environment, not scripts that run in the CI environment (all scripts are potentials for running in CI).

File: scripts/src/javatests/org/oppia/android/scripts/ci/LicenseTextsCheckTest.kt
Comment (BenHenning) [inline scripts/src/javatests/org/oppia/android/scripts/ci/LicenseTextsCheckTest.kt]: Shouldn't this case fail? Perhaps we should verify that the XML file itself is empty, too (e.g. by using an XML parser like we do in the accessibility label script & verify that it has no elements--would that work?).

File: scripts/src/javatests/org/oppia/android/scripts/ci/LicenseTextsCheckTest.kt
Comment (BenHenning) [inline scripts/src/javatests/org/oppia/android/scripts/ci/LicenseTextsCheckTest.kt]: Another test worth including, I think: running the script when the XML file doesn't exist (to cover cases when it's moved in the future).

**Review by BenHenning — CHANGES_REQUESTED** [review]
Thanks @prayutsu. Did a full review & had a few comments--PTAL.

File: scripts/src/java/org/oppia/android/scripts/ci/LicenseTextsCheck.kt
Comment (prayutsu) [reply scripts/src/java/org/oppia/android/scripts/ci/LicenseTextsCheck.kt]: Done.

File: scripts/src/java/org/oppia/android/scripts/ci/LicenseTextsCheck.kt
Comment (prayutsu) [reply scripts/src/java/org/oppia/android/scripts/ci/LicenseTextsCheck.kt]: Done.

File: scripts/src/java/org/oppia/android/scripts/ci/LicenseTextsCheck.kt
Comment (prayutsu) [reply scripts/src/java/org/oppia/android/scripts/ci/LicenseTextsCheck.kt]: Done.

File: scripts/src/java/org/oppia/android/scripts/ci/LicenseTextsCheck.kt
Comment (prayutsu) [reply scripts/src/java/org/oppia/android/scripts/ci/LicenseTextsCheck.kt]: Done.

File: scripts/src/java/org/oppia/android/scripts/ci/LicenseTextsCheck.kt
Comment (prayutsu) [reply scripts/src/java/org/oppia/android/scripts/ci/LicenseTextsCheck.kt]: Done.

File: scripts/src/java/org/oppia/android/scripts/ci/LicenseTextsCheck.kt
Comment (prayutsu) [reply scripts/src/java/org/oppia/android/scripts/ci/LicenseTextsCheck.kt]: Done.

File: scripts/src/javatests/org/oppia/android/scripts/ci/LicenseTextsCheckTest.kt
Comment (prayutsu) [reply scripts/src/javatests/org/oppia/android/scripts/ci/LicenseTextsCheckTest.kt]: Done.

File: scripts/src/javatests/org/oppia/android/scripts/ci/LicenseTextsCheckTest.kt
Comment (prayutsu) [reply scripts/src/javatests/org/oppia/android/scripts/ci/LicenseTextsCheckTest.kt]: I think the XML file will contain dummy texts so this test case should pass or am I missing something here?

Comment (BenHenning) [reply scripts/src/javatests/org/oppia/android/scripts/ci/LicenseTextsCheckTest.kt]: I think if you generated the file & manually readded the comment, it would cause the check to pass. Technically, that scenario should fail. While the comment acts as a canary, it's fairly easy to break it. The additional check that the XML file is also logically empty I think helps make this check much more robust overall.

File: scripts/src/java/org/oppia/android/scripts/license/BUILD.bazel
Comment (BenHenning) [inline scripts/src/java/org/oppia/android/scripts/license/BUILD.bazel:42]: Do you also need to update the root scripts build file to point to the correct new location?

**Review by BenHenning — COMMENTED** [review]
Thanks @prayutsu. Just one new comment, and I followed up on one of the open comment threads.

File: scripts/src/java/org/oppia/android/scripts/license/BUILD.bazel
Comment (prayutsu) [reply scripts/src/java/org/oppia/android/scripts/license/BUILD.bazel:42]: Done.

File: scripts/src/javatests/org/oppia/android/scripts/ci/LicenseTextsCheckTest.kt
Comment (prayutsu) [reply scripts/src/javatests/org/oppia/android/scripts/ci/LicenseTextsCheckTest.kt]: Discussed on chat.

File: .github/CODEOWNERS
Comment (BenHenning) [inline .github/CODEOWNERS]: Does this need to go into the overrides section since app's XML files are blanket codeowner assigned below?

File: .github/CODEOWNERS
Comment (BenHenning) [inline .github/CODEOWNERS]:
```suggestion
# License texts.
```

**Review by BenHenning — COMMENTED** [review]
Thanks @prayutsu. Just 2 small comments otherwise the PR LGTM.

File: .github/CODEOWNERS
Comment (prayutsu) [reply .github/CODEOWNERS]: Done.

File: .github/CODEOWNERS
Comment (prayutsu) [reply .github/CODEOWNERS]: Moved to overrides section

**Review by BenHenning — APPROVED** [review]
Thanks @prayutsu! LGTM.

**Comment by prayutsu** [comment]
@BenHenning Assigning the PR to you, so that you can merge it after you merge #3520.

**Review by BenHenning — CHANGES_REQUESTED** [review]
Thanks @prayutsu. Marking as request changes per your comment. Will double check & re-approve once #3520 is merged.

**Comment by BenHenning** [comment]
@prayutsu please bring this up-to-date with develop now that its based PR is merged & assign back.

**Comment by prayutsu** [comment]
@BenHenning PTAL.

File: scripts/src/javatests/org/oppia/android/scripts/license/LicenseTextsCheckTest.kt
Comment (BenHenning) [inline scripts/src/javatests/org/oppia/android/scripts/license/LicenseTextsCheckTest.kt]: Nit: we should actually throw a custom error exception (e.g. with ``check(file.exists()) { "File does not exist: $file" }``) since the exact "No such file or directory" can change between JVM implementations.

**Review by BenHenning — COMMENTED** [review]
Thanks @prayutsu! Just noticed one thing that I missed earlier, but otherwise this LGTM.

File: scripts/src/javatests/org/oppia/android/scripts/license/LicenseTextsCheckTest.kt
Comment (prayutsu) [reply scripts/src/javatests/org/oppia/android/scripts/license/LicenseTextsCheckTest.kt]: Done.

**Comment by prayutsu** [comment]
@BenHenning PTAL.

**Review by BenHenning — APPROVED** [review]
Thanks @prayutsu. LGTM!

Can't enable auto-merge since I need to verify that the test change is passing (Bazel tests do not currently block PR submission).

**Comment by BenHenning** [comment]
@prayutsu please assign back when CI is passing or merge yourself (but only if all CI checks are passing since some, such as the Bazel tests, are currently not required per our GitHub config).

**Pull request merged**
